src/ralph/backends/data/mongo.py

```diff
@@ -0,0 +1,399 @@
+            if "timestamp" not in statement:
+                msg = f"statement {statement} has no 'timestamp' field"
+                if ignore_errors:
+                    logger.warning(msg)
+                    continue
+                raise BadFormatException(msg)
+            try:
+                timestamp = int(isoparse(statement["timestamp"]).timestamp())
+            except ValueError as err:
+                msg = f"statement {statement} has an invalid 'timestamp' field"
+                if ignore_errors:
+                    logger.warning(msg)
+                    continue
+                raise BadFormatException(msg) from err
+            document = {
+                "_id": ObjectId(
+                    # This might become a problem in February 2106.
+                    # Meanwhile, we use the timestamp in the _id field for pagination.
+                    struct.pack(">I", timestamp)
+                    + bytes.fromhex(
+                        hashlib.sha256(bytes(statement.get('id', str(uuid4())), "utf-8")).hexdigest()[:16]
+                    )
+                ),
+                "_source": statement,
+            }
+
+            yield document
+
+    def bulk_import(self, batch: list, ignore_errors: bool = False):
+        """Inserts a batch of documents into the selected database collection."""
+        try:
+            new_documents = self.collection.insert_many(batch)
+        except BulkWriteError as error:
+            if not ignore_errors:
+                raise BackendException(
+                    *error.args, f"{error.details['nInserted']} succeeded writes"
+                ) from error
+            logger.warning(
+                "Bulk importation failed for current documents chunk but you choose "
+                "to ignore it.",
+            )
+            return error.details["nInserted"]
+
+        inserted_count = len(new_documents.inserted_ids)
+        logger.debug("Inserted %d documents chunk with success", inserted_count)
+
+        return inserted_count
+    
+    def bulk_delete(self, batch: list, ignore_errors: bool = False):
+        """Delete a batch of documents into the selected database collection."""
+        try:
+            new_documents = self.collection.delete_many({"_source.id":{'$in':batch}})
+        except BulkWriteError as error:
+            if not ignore_errors:
+                raise BackendException(
+                    *error.args, f"{error.details['nInserted']} succeeded writes"
+                ) from error
+            logger.warning(
+                "Bulk delete failed for current documents chunk but you choose "
+                "to ignore it.",
+            )
+            return error.details["nInserted"]
+
+        deleted_count = new_documents.deleted_count
+        logger.debug("Deleted %d documents chunk with success", deleted_count)
+
+        return deleted_count
+
+    def write(
+        self,
+        data: Union[IOBase, Iterable[bytes], Iterable[dict]],
```

Comment:
Not sure if `IOBase` should be supported. Are you able to parse bytes from an IOBase to a dict?

Reply:
You're right, but you should probably test it to see if this works